### PR TITLE
Dockerise tests

### PR DESCRIPTION
**Pull request by PhysicsJon**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -23,65 +23,59 @@ variables:
...
-  enabled: false
-  inputs:
-    PathtoPublish: '$(Build.ArtifactStagingDirectory)'
-    ArtifactName: 'drop'
-    publishLocation: 'Container'
-      
-- bash: 
-    "curl -k -i $(PBURL) --fail --connect-timeout 30"
+- bash: |
+    echo "##vso[task.setvariable variable=PBHOSTURL]$(echo "${PBURL#'https://'}")"
```
Comment (talawson05) [inline azure-pipelines.yml:41]: are you expecting the value to be without https://?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -23,65 +23,59 @@ variables:
...
-    cat domain.yml
-    cat /etc/hosts
-  displayName: 'Resolve domain'
-- task: UseDotNet@2
-  displayName: 'Use DotNet Core 3.1'
+- task: UseGitVersion@5
+  name: gitversion
+  displayName: Work out version
   inputs:
-    version: '3.1.101'
```
Comment (talawson05) [inline azure-pipelines.yml:52]: @PhysicsJon are we specifying the dotnet version in the solution?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -23,65 +23,59 @@ variables:
...
+    do
+      status=$(docker inspect --format "{{json .State.Health.Status }}" "$firstNode" | tr -d '"')
+      
+      if [ "$status" = "healthy" ]; then echo "Selenium grid is ready" && exit 0; fi
+      n=$((n+1)) 
+      sleep 1
+    done
+
+    echo "Selenium grid cannot access $PBHOSTURL"
+    exit 1
```
Comment (talawson05) [inline azure-pipelines.yml]: if this works we should put this in a file and run it from there

File: azure-pipelines.yml [hunk 4]
```diff
@@ -23,65 +23,59 @@ variables:
...
-    
+
     dotnet test src/NHSDPublicBrowseAcceptanceTests.Tests/*.csproj -v n
-  name: runTests
-  displayName: Run Acceptance Tests
\ No newline at end of file
+  displayName: Run Acceptance Tests
+  
+- task: Docker@2
+  condition: and(always(), eq(variables['Build.SourceBranch'], 'refs/heads/development'))
```
Comment (talawson05) [inline azure-pipelines.yml:62]: can we have a displayName on this task as i'm not sure what it's doing

File: azure-pipelines.yml (hunk 2 above)
Comment (PhysicsJon) [reply azure-pipelines.yml:52]: Nope, we should be grabbing latest (the specflow fix is in the variables set when building the image)

File: azure-pipelines.yml (hunk 1 above)
Comment (PhysicsJon) [reply azure-pipelines.yml:41]: PBURL has https://, we need to remove that for the hosts file config addition

Comment (adrian-fitsios) [reply azure-pipelines.yml:41]: No, I think that the PBURL should include the protocol.

The tests should have the protocol, but the entry in /etc/hosts file shouldn't have it.

File: azure-pipelines.yml (hunk 3 above)
Comment (adrian-fitsios) [reply azure-pipelines.yml]: Sure thing, I'll put it in its own script.

File: azure-pipelines.yml (hunk 2 above)
Comment (adrian-fitsios) [reply azure-pipelines.yml:52]: Put the task back, we're now grabbing latest patch version from 3.1.x

**Review by stuart-bass-bjss — APPROVED** [review]

**Review by talawson05 — APPROVED** [review]

**Review by talawson05 — APPROVED** [review]

**Pull request merged**